x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/ScriptBooleanMappedFieldType.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.runtimefields.mapper;
+
+import org.apache.lucene.search.MatchNoDocsQuery;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.common.lucene.search.Queries;
+import org.elasticsearch.common.time.DateMathParser;
+import org.elasticsearch.index.mapper.BooleanFieldMapper;
+import org.elasticsearch.index.query.QueryShardContext;
+import org.elasticsearch.script.Script;
+import org.elasticsearch.search.DocValueFormat;
+import org.elasticsearch.search.lookup.SearchLookup;
+import org.elasticsearch.xpack.runtimefields.BooleanScriptFieldScript;
+import org.elasticsearch.xpack.runtimefields.fielddata.ScriptBooleanFieldData;
+import org.elasticsearch.xpack.runtimefields.query.BooleanScriptFieldExistsQuery;
+import org.elasticsearch.xpack.runtimefields.query.BooleanScriptFieldTermQuery;
+
+import java.time.ZoneId;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+
+public class ScriptBooleanMappedFieldType extends AbstractScriptMappedFieldType {
+    private final BooleanScriptFieldScript.Factory scriptFactory;
+
+    ScriptBooleanMappedFieldType(String name, Script script, BooleanScriptFieldScript.Factory scriptFactory, Map<String, String> meta) {
+        super(name, script, meta);
+        this.scriptFactory = scriptFactory;
+    }
+
+    @Override
+    protected String runtimeType() {
+        return BooleanFieldMapper.CONTENT_TYPE;
+    }
+
+    @Override
+    public Object valueForDisplay(Object value) {
+        throw new UnsupportedOperationException();
```

Comment:
why do we throw? Shouldn't we borrow the logic from https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/index/mapper/BooleanFieldMapper.java#L157 ?

Reply:
I can do that, but I don't think it is ever called!

Reply:
oh well I did not realize it was unused, in that case I am good with throwing, but add a comment?